src/core.jl

```diff
@@ -314,8 +311,30 @@ julia> common_neighbors(g, 1, 4)
  3
 ```
 """
-common_neighbors(g::AbstractGraph, u::Integer, v::Integer) =
-    intersect(neighbors(g, u), neighbors(g, v))
+function common_neighbors(g::AbstractGraph, u::Integer, v::Integer)
+    common = Vector{Int}()
+    u_nbrs, v_nbrs = neighbors(g, u), neighbors(g, v)
+    i, j = 1, 1
+    u_len, v_len = length(u_nbrs), length(v_nbrs)
+
+    if (u_nbrs[u_len] < v_nbrs[1]) || (u_nbrs[1] > v_nbrs[v_len])
```

Comment:
If you write `u_nbrs[end]` instead of `u_nbrs[u_len]`, the intention might be a bit easier to see.